backend/btrixcloud/storages.py

```diff
@@ -24,462 +25,497 @@
     sync_get_log_stream,
 )
 
+from .utils import is_bool
+
 
 CHUNK_SIZE = 1024 * 256
 
 
 # ============================================================================
-def init_storages_api(org_ops, crawl_manager, user_dep):
-    """API for updating storage for an org"""
+class StorageOps:
+    """All storage handling, download/upload operations"""
 
-    router = org_ops.router
-    org_owner_dep = org_ops.org_owner_dep
+    def __init__(self, org_ops, crawl_manager):
+        self.storages = {}
+
+        self.org_ops = org_ops
+        self.crawl_manager = crawl_manager
+
+        self.is_local_minio = is_bool(os.environ.get("IS_LOCAL_MINIO"))
+
+        with open("/tmp/storages/storages.json", encoding="utf-8") as fh:
+            storage_list = json.loads(fh.read())
+
+        for storage in storage_list:
+            name = storage.get("name")
+            type_ = storage.get("type", "s3")
```

Comment:
I wonder if we should include a `default` or `primary` boolean field to distinguish the primary storage from replicas more consistently than by name.